File: app/src/main/java/br/com/gabryel/financeapp/investment/SimpleMovement.java
package br.com.gabryel.financeapp.investment;

import org.threeten.bp.LocalDate;

/**
 * Simple implementation of a movement.
 * <p/>
 * Created by gabryel on 04/04/16.
 */
public class SimpleMovement implements Movement {

	private final LocalDate movementDate;

	private final double value;

	public SimpleMovement(LocalDate movementDate, Movement... movements) {
		this.movementDate = movementDate;
		double partialValue = 0;

		for (Movement mov : movements) {
			partialValue += mov.getValue();
		}

		this.value = partialValue;
	}

	public SimpleMovement(LocalDate movementDate, double value) {
		this.movementDate = movementDate;
		this.value = value;
	}

	@Override
	public LocalDate getMovementDate() {
		return movementDate;
	}

	@Override
	public double getValue() {
		return value;
	}
}

File: app/src/main/java/br/com/gabryel/financeapp/investment/SimpleMovement.kt
package br.com.gabryel.financeapp.investment

import org.threeten.bp.LocalDate

/**
 * Simple implementation of a movement.
 *
 * Created by gabryel on 04/04/16.
 */
class SimpleMovement : Movement {

    override val movementDate: LocalDate

    override val value: Double

    constructor(movementDate: LocalDate, vararg movements: Movement) {
        this.movementDate = movementDate
        var partialValue = 0.0

        for (mov in movements) {
            partialValue += mov.value
        }

        this.value = partialValue
    }

    constructor(movementDate: LocalDate, value: Double) {
        this.movementDate = movementDate
        this.value = value
    }
}
